book/src/user/experimental-features.md

```diff
@@ -0,0 +1,138 @@
+
+To support different kind of polynomial commitment scheme, we added a trait `CommitmentScheme` to allow create/verify proof with different commitment scheme implementation, currently there are 2 available implementations in this fork:
+
+- `IPACommitmentScheme`
+
+  The original implementation from `zcash/halo2` with the original multi-open strategy `{Prover,Verifier}IPA`
+
+- `KZGCommitmentScheme`
+
+  KZG commitment scheme as in [GWC19](https://eprint.iacr.org/2019/953), which doesn't commit the instance columns, with 2 multi-open strategies available:
+
+  - `{Prover,Verifier}GWC` - The original strategy in [GWC19](https://eprint.iacr.org/2019/953)
+  - `{Prover,Verifier}SHPLONK` - The strategy proposed in [BDFG20](https://eprint.iacr.org/2020/081)
+
+When using `create_proof` and `verify_proof`, we need to specify the commitment scheme and multi-open strategy like:
+
+```rust
+// Using IPA
+create_proof<IPACommitmentScheme<_>, ProverIPA<_>, _, _, _, _>
+verify_proof<IPACommitmentScheme<_>, ProverIPA<_>, _, _, _>
+
+// Using KZG with GWC19 mutli-open strategy
+create_proof<KZGCommitmentScheme<_>, ProverGWC<_>, _, _, _, _>
+verify_proof<KZGCommitmentScheme<_>, ProverGWC<_>, _, _, _>
+
+// Using KZG with BDFG20 mutli-open strategy
+create_proof<KZGCommitmentScheme<_>, ProverSHPLONK<_>, _, _, _, _>
+verify_proof<KZGCommitmentScheme<_>, ProverSHPLONK<_>, _, _, _>
+```
+
+## `ConstraintSystem` extension
+
+### Dynamic lookup
+
+`ConstraintSystem::lookup_any` is added for use cases that need to lookup dynamic table instead of fixed table.
+
+Unlike `ConstraintSystem::lookup` which only allows `TableColumn`(s) as table, it allows any `Expression`(s) without simple selector.
+
+### Shuffle
+
+`ConstraintSystem::shuffle` is added for use cases that only need shuffle without pre-defined mapping.
+
+It allows us to prove any `Expression`(s) without simple selector is shuffled from the other.
+
+### Multi-phase
+
+`ConstraintSystem::advice_column_in` and `ConstraintSystemchallenge_usable_after` are added for use cases that build PIOP sub-routine themselves, currently it supports up-to 3 phases as `{First,Second,Third}Phase`.
+
+It allows us to allocate advice column in different interactive phases with extra challenges squeezed in-between. For example in `halo2_proofs/examples/shuffle.rs` it shows how to build a customized shuffle argument with such API.
+
+### Unblinded advice column
+
+`ConstraintSystem::unblinded_advice_column` is added for use cases that want to re-use advice column commitment among different proofs.
+
+Worth mentioning, re-using advice column commitment in different proofs will need more blinding factors than the amount that prover adds, otherwise some information will be leaked and it's no longer perfect zero-knowledge.
+
+## `Expression` extension
+
+- `Expression::Challenge`
+
+  A variant added for multi-phase. It can be obtained by `ConstraintSystemchallenge_usable_after` and used as a pseudo-random constant.
```

Comment:
```suggestion
  A variant added for multi-phase. It can be obtained by [`ConstraintSystemChallenge::challenge_usable_after`](https://github.com/privacy-scaling-explorations/halo2/blob/9b33f9ce524dbb9133fc8b9638b2afd0571659a8/halo2_proofs/src/plonk/circuit.rs#L2177-L2192) and used as a pseudo-random constant.
```